bootstrap/pkg/kfapp/coordinator/coordinator.go

```diff
@@ -300,6 +305,8 @@ func LoadKfAppCfgFile(cfgfile string) (kftypesv3.KfApp, error) {
 		}
 	}
 
+	// Set the ConfigFileName for kfdef. This is used when we WriteToConfig.
+	kfdef.Spec.ConfigFileName = filepath.Base(appFile)
```

Comment:
I'm not sure I understand why we need a new field `ConfigFileName`?  just like what you do here, we could always separate the values we need with `Base` and `Dir`, no?

Reply:
We need to save it so when we write out, we know the filename